### PR TITLE
[Refactoring] New extension structure

### PR DESCRIPTION
**Pull request by PedroTroller**
@akovalyov What do you think of this implementation ?

### CODE REVIEW

**Comment by akovalyov** [comment]
Like. :+1:

File: src/Knp/DictionaryBundle/Twig/DictionaryExtension.php [hunk 1]
```diff
@@ -7,16 +7,16 @@
 class DictionaryExtension extends \Twig_Extension
 {
     /**
-     * @var DictionaryRegistry
+     * @var Dictionarydictionaries
```
Comment (Koc) [inline src/Knp/DictionaryBundle/Twig/DictionaryExtension.php]: smth wring with vdoc here

**Comment by Koc** [comment]
What sense of adding filter? Filters usually transforms values.

**Comment by Koc** [comment]
```twig
{{ dictionary('my_key', 'dictionary_name') }}
{{ dictionary('dictionary_name') }}
```
Looks inconsistent. Why key name is not the second argument?

File: src/Knp/DictionaryBundle/Twig/DictionaryExtension.php (hunk 1 above)
Comment (PedroTroller) [inline src/Knp/DictionaryBundle/Twig/DictionaryExtension.php]: Indeed

**Comment by PedroTroller** [comment]
For example, if your key is stored into an object (Doctrine model for example), you can use `{{ my_model.civility|dictionary('gender') }}`.

For function, the logic is `{{ dictionary('what_you_are_searching_form', 'into_what') }}`

**Comment by Koc** [comment]
Why `dictionary('gender', my_model.civility)` not enough?

**Comment by PedroTroller** [comment]
@Koc I think this one is much better ;)

https://github.com/KnpLabs/DictionaryBundle/pull/13

**Pull request merged**